### ISSUE TITLE
Show more explicit error when the canvas size exceeds browser size

### ISSUE THREAD

**Issue by kanitw** [vega/vega#1522]
https://stackoverflow.com/questions/54226669/one-more-record-and-the-chart-disappears-is-there-a-limit-in-vl

Currently there is no error and this can confuse users.

**Comment by jheer** [comment]
Yes, this would be a nice feature. However, canvas size limitations vary by browser, device, and over the course of time. Browsers do not provide a direct means of requesting the maximum size, and have different policies for handling over-sized rasters (e.g., ignore all drawing commands without errors, or draw in smaller usable space only).

So sadly I do not yet know a consistent and efficient way to do this!

More information here: https://stackoverflow.com/questions/6081483/maximum-size-of-a-canvas-element

**Comment by danmarshall** [comment]
Just found this library that determines maximum canvas size: https://github.com/jhildenbiddle/canvas-size#canvas-size

**Comment by corneliusroemer** [comment]
To help others find this issue, Chrome displays a sad face on the canvas when this happens. Would be great if browsers reported something in console logs as well, but they sadly don't seem to.
<img width="578" alt="image" src="https://github.com/vega/vega/assets/25161793/8162ec2c-6356-4088-b3f6-7d342f1aec00">
